### PR TITLE
Mr15 implement robot upgrading

### PR DESCRIPTION
**Pull request by Chris-TH**

### CODE REVIEW

**Comment by DennisWae** [comment]
Please don't put the code of other tickets into unrelated pull requests, as this makes them harder to review. Please revert your previoues two commits and create new branches for them and open new pull requests.

**Review by DennisWae — CHANGES_REQUESTED** [review]
Please revert your previous two commits

**Review by DennisWae — CHANGES_REQUESTED** [review]
Wait until MR8 ist merged into main and then merge main into this branch.

**Comment by DennisWae** [comment]
MR8 is now merged.

**Comment by Chris-TH** [comment]
Merged branch 'main' into MR15-Implement-Robot-Upgrading

File: src/main/kotlin/com/msd/robot/application/RobotApplicationService.kt
Comment (DennisWae) [inline src/main/kotlin/com/msd/robot/application/RobotApplicationService.kt:136]: You need to persist the Robot after modifying it.

**Review by DennisWae — CHANGES_REQUESTED** [review]

**Review by DennisWae — APPROVED** [review]

**Pull request merged**
